include/visitors/interpreter.h

```diff
@@ -195,16 +281,35 @@ class Interpreter : public Visitor
         auto expr = this->stack[this->stack.size() - 1];
         this->stack.pop_back();
 
-        this->stack.push_back(-expr);
+        this->stack.push_back(Value(DataType::INT, variant(-std::get<int>(expr.data))));
```

Comment:
Can you add some error handling for if this is not an int? Also, can you add floats to this?